sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicyTests.java

```diff
@@ -3,144 +3,203 @@
+        assertEquals("user-agent", httpAttributes.get("http.user_agent"));
+        assertEquals("foo", httpAttributes.get(AZ_TRACING_NAMESPACE_KEY));
+        assertEquals(201L, httpAttributes.get("http.status_code"));
+        assertEquals(X_MS_REQUEST_ID, httpAttributes.get("x-ms-request-id"));
     }
 
-    private static HttpPipeline createHttpPipeline() {
-        final HttpClient httpClient = HttpClient.createDefault();
-        final List<HttpPipelinePolicy> policies = new ArrayList<>();
-        HttpPolicyProviders.addAfterRetryPolicies(policies);
+    private Map<String, Object> getAttributes(SpanData span) {
+        Map<String, Object> attributes = new HashMap<>();
+        span.getAttributes().forEach((k, v) -> attributes.put(k.getKey(), v));
 
+        return attributes;
+    }
+
+    private static HttpPipeline createHttpPipeline(Tracer tracer) {
         final HttpPipeline httpPipeline = new HttpPipelineBuilder()
-            .httpClient(httpClient)
-            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(new SimpleMockHttpClient())
+            .policies(new OpenTelemetryHttpPolicy(tracer))
             .build();
         return httpPipeline;
     }
 
-    private static SpanContext getNonRemoteSpanContext(String diagnosticId) {
-        Context updatedContext = AmqpPropagationFormatUtil.extractContext(diagnosticId, Context.NONE);
-        SpanContext spanContext = (SpanContext) updatedContext.getData(SPAN_CONTEXT_KEY).get();
-        return SpanContext.create(spanContext.getTraceId(), spanContext.getSpanId(),
-            spanContext.getTraceFlags(), spanContext.getTraceState());
-    }
+    private static class SimpleMockHttpClient implements HttpClient {
+
+        @Override
+        public Mono<HttpResponse> send(HttpRequest request) {
+            HttpHeaders headers = new HttpHeaders()
+                .set("Content-Type", "application/json")
+                .set("x-ms-request-id", X_MS_REQUEST_ID);
 
-    private static void verifySpanContextAttributes(SpanContext expectedSpanContext, SpanContext actualSpanContext) {
-        assertEquals(expectedSpanContext.getTraceId(), actualSpanContext.getTraceId());
-        assertNotEquals(expectedSpanContext.getSpanId(), actualSpanContext.getSpanId());
-        assertEquals(expectedSpanContext.getTraceFlags(), actualSpanContext.getTraceFlags());
-        assertEquals(expectedSpanContext.getTraceState(), actualSpanContext.getTraceState());
-        assertEquals(expectedSpanContext.isValid(), actualSpanContext.isValid());
-        assertEquals(expectedSpanContext.isRemote(), actualSpanContext.isRemote());
+            HttpResponse response = new MockHttpResponse(request, headers);
+            return Mono.just(response);
+        }
     }
 
-    /**
-     * Maps to the JSON return values from http://httpbin.org.
-     */
-    static class HttpBinJSON {
+    private static class MockHttpResponse extends HttpResponse {
```

Comment:
I believe there is a MockHttpResponse class in azure-core-test which covers this use case